src/Components/test/E2ETest/Tests/VirtualizationTest.cs

```diff
@@ -204,5 +210,136 @@ public void CanUseViewportAsContainer()
             // Validate that the top spacer has expanded.
             Browser.NotEqual(expectedInitialSpacerStyle, () => topSpacer.GetAttribute("style"));
         }
+
+        [Fact]
+        public void CanMutateDataInPlace_Sync()
+        {
+            Browser.MountTestComponent<VirtualizationDataChanges>();
+
+            // Initial data
+            var container = Browser.FindElement(By.Id("using-items"));
+            Browser.Collection(() => GetPeopleNames(container),
+                name => Assert.Equal("Person 1", name),
+                name => Assert.Equal("Person 2", name),
+                name => Assert.Equal("Person 3", name));
+
+            // Mutate one of them
+            var itemToMutate = container.FindElements(By.ClassName("person"))[1];
+            itemToMutate.FindElement(By.TagName("button")).Click();
+
+            // See changes
+            Browser.Collection(() => GetPeopleNames(container),
+                name => Assert.Equal("Person 1", name),
+                name => Assert.Equal("Person 2 MUTATED", name),
+                name => Assert.Equal("Person 3", name));
+        }
+
+        [Fact]
+        public void CanMutateDataInPlace_Async()
+        {
+            Browser.MountTestComponent<VirtualizationDataChanges>();
+
+            // Initial data
+            var container = Browser.FindElement(By.Id("using-itemsprovider"));
```

Comment:
nit: since the tests are named sync/async, maybe the element id's should also have sync/async suffixes to align, mostly to make it more clear from just looking at the test code what's going on

Reply:
Unless you feel strongly, I'd prefer to keep this as-is, because it's not really about sync vs async in terms of the underlying scenario. The underlying scenario is really about `Items` vs `ItemsProvider`. I know the existing E2E test cases are named with sync/async which is why I followed that pattern there, but the implementation doesn't really need to do that.